### PR TITLE
JavaScript: Add Sets concept

### PR DESCRIPTION
**Pull request by Javokhir12**
### Description

* Add `Sets` concept in `JavaScript`

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by yangc95** [comment]
🎄 First review underway, will 🏷️ you once complete and ready for you to look over 👍

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:3]:
```suggestion
Description: 'A set is a special object that stores unique values without a key.'
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:16]:
```suggestion
A set is an object in JavaScript that is a collection of unique values.
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:18]:
```suggestion
A `Set` object can be created with the `Set()` constructor function.
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:32]:
```suggestion
## Adding value to a `Set`
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:40]:
```suggestion
## Determining whether a value is in a `Set`
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:37]:
```suggestion
set.add(1); // Still Set { 1, 'text' } because 1 is already in the set
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:48]:
```suggestion
## Retrieving the size of a `Set`
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:56]:
```suggestion
## Removing values from a `Set`
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:66]:
```suggestion
## Removing everything from a `Set`
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:73]:
```suggestion
## Initializing a `Set` using iterables (e.g. arrays)
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:86]: Let's shift this explanation to above this codebyte 👍

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:30]:
```suggestion
- `set.add(value)`: Adds a value to the set. If the value is already in the set, it doesn't do anything.
- `set.delete(value)`: Removes the given value, returns `true` if the value existed in the set, otherwise `false`.
- `set.has(value)`: Returns `true` if the value exists in the set, otherwise `false`.
- `set.clear()`: Deletes everything from the set.
- `set.size`: Returns the number of elements in the set.
```

Just a style preference 📝

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:36]:
```suggestion
set.add(1); // Set { 1 }
set.add('text'); // Set { 1, 'text' }
```

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [reply content/javascript/concepts/sets/sets.md:40]: Let's also add some preface for the code block below

File: content/javascript/concepts/sets/sets.md
Comment (yangc95) [inline content/javascript/concepts/sets/sets.md:54]: Preference for the variable used throughout this entry:

- Rather than just `set` which can get confusing when we are referring to a `Set` object, `mySet` is a clearer alternative name

**Review by yangc95 — COMMENTED** [review]
🎁  @Javokhir12 thanks for your entry ☺️ Review 1️⃣  is complete, please 🏷️ me once you've taken a look!

**Review by yangc95 — APPROVED** [review]
LGTM ! Thanks for re-requesting a review from me @Javokhir12 

This is ready for review 2️⃣

File: content/javascript/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/javascript/concepts/sets/sets.md:16]:
```suggestion
A `Set` is an object in JavaScript that is a collection of unique values.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
Looks good @Javokhir12, @yangc95. I only had one suggestion.

**Comment by yangc95** [comment]
> Looks good @Javokhir12, @yangc95. I only had one suggestion.

☺️ Thanks @SSwiniarski ! I'll merge this once you approve ✅

**Review by SSwiniarski — APPROVED** [review]

**Comment by SSwiniarski** [comment]
@yangc95, all set to go.

**Comment by yangc95** [comment]
Thanks @Javokhir12 🥳 

Live here: https://www.codecademy.com/resources/docs/javascript/sets

Although, the contributor's panel is coming up as anonymous. I recall seeing your contributor image pop up for another entry so I feel your GitHub/Codecademy settings may not be the cause for this. We've been seeing this issue across entries and I'll add this one to the pile.

cc: @SSwiniarski @Dusch4593 @christian-dinh

**Comment by Dusch4593** [comment]
Happy New Year @Javokhir12! We have an update!

In addition to setting your Codecademy profile visibility to "Everyone", you have to enable SAML single sign-on ([SSO](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/about-authentication-with-saml-single-sign-on)) between your Codecademy and GitHub profiles _through_ the same primary email address. This can be done by following some steps [here in the CONTRIBUTING.md guide](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#codecademy-username).

Hope this helps but definitely let us know if you have any questions! 😃

**Comment by Javokhir12** [comment]
> Hi, @Dusch4593 . Happy New Year to you too. Thanks for letting me know, I will definitely look into this.

**Pull request merged**
